.github/workflows/relay.yml

```
@@ -46,6 +46,9 @@ jobs:
       - name: Run unit tests
         run: nix develop -c go test -v ./pkg/terra/...
 
+      - name: Run tests with the race detector enabled
+        run: nix develop -c go test -v ./pkg/terra/... -race -count=10
```

Could consider using the `GORACE` env var to isolate the failures from the verbose test output:
```suggestion
        run: GORACE="log_path=$PWD/race" nix develop -c go test -v ./pkg/terra/... -race -count=10
```